src/battle_ai_util.c

```diff
@@ -2616,7 +2616,7 @@ bool32 AI_CanPutToSleep(u8 battlerAtk, u8 battlerDef, u16 defAbility, u16 move,
 bool32 AI_CanBePoisoned(u8 battler, u16 ability)
 {
     if (ability == ABILITY_IMMUNITY
-      || ability == ABILITY_PASTEL_VEIL
+      || IsAbilityOnSide(battler, ABILITY_PASTEL_VEIL)
```

Comment:
`IsAbilityOnSide` will need its own AI version, e.g. `AI_IsAbility that calls `AI_GetAbility`, because the AI doesn't necessarily know the players abilities. This isn't necessarily needed in this PR, especially because I'm seeing other references to `IsAbilityOnSide` in src/battle_ai_util.c, but will otherwise need an issue opened.

Reply:
I think it would be best to resolve this in another PR, I've opened an issue for it.